Example/DipSampleApp/DependencyContainers.swift

```diff
@@ -27,15 +27,17 @@ let wsDependencies = DependencyContainer<WebService>() { dip in
 }
 
 
+let NamedStarshipProviderTag: Tag    = "NamedStarshipProvider"
+let PlistPersonProviderTag: Tag      = "PlistPersonProvider"
 
 // MARK: Dependency Container for Providers
-let providerDependencies = DependencyContainer<Int>() { dip in
+let providerDependencies = DependencyContainer() { dip in
     
     if FAKE_PERSONS {
         
         // 1) Register the PersonProviderAPI singleton, one generic and one specific for a specific personID
         dip.register(instance: DummyPilotProvider() as PersonProviderAPI)
-        dip.register(0, instance: PlistPersonProvider(plist: "mainPilot") as PersonProviderAPI)
+        dip.register(PlistPersonProviderTag, instance: PlistPersonProvider(plist: "mainPilot") as PersonProviderAPI)
```

Comment:
By changing that, you associated that `PersonProviderAPI` with a `String` tag instead of an `Int` tag. This means that the example project probably won't work as expected anymore, since when resolving, you kept [that](https://github.com/AliSoftware/Dip/pull/3/files#diff-ca934cf6a8bc3ae3d5d8e7afbcd16082R20) logic:

```swift
let provider = providerDependencies.resolve(.Int(personID)) as PersonProviderAPI
```

Probably better rollback that line to `dip.register(0, …)` then?